.github/workflows/application_settings-buid_deploy_docs_release.yml

```diff
@@ -0,0 +1,79 @@
+# Simple workflow for deploying static content to GitHub Pages
+# Create versioned docs on releas.
+name: "Build and deploy versioned docs on release"
+
+on:
+  release:
+    types: [published]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  id-token: write
+
+# Allow one concurrent deployment
+concurrency:
+  group: "pages"
+  cancel-in-progress: true
+
+jobs:
+  build_and_deploy:
+    # environment: dev_environment
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@master
+        with:
+          fetch-depth: 0 # otherwise, you will failed to push refs to dest repo
+          ref: release
+
+      - name: Set env
+        run: echo "RELEASE_VERSION=${GITHUB_REF#refs/*/}" >> $GITHUB_ENV
+
+      - name: Setup python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.11
+
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.2.2
+          virtualenvs-create: true
+          virtualenvs-in-project: true
+
+      - name: Check lockfile
+        run: poetry lock --check
+
+      - name: Install packages
+        run: poetry install --no-interaction --with docs
+
+      - name: Build html
+        run:
+          poetry run portray as_html --overwrite -o site
+
+      - name: Setup git config
+        run: |
+          git config user.name "GitHub Actions Bot"
+          git config user.email "<>"
+
+      - name: Rename folder and redirect stable to new release
+        run: |
+            echo "Renaming site folder and committing"
+            git checkout gh-pages
+            mv site "$RELEASE_VERSION"
+            git add "$RELEASE_VERSION"/
+            git commit -m "add docs for new version $RELEASE_VERSION"
+            echo "Redirecting stable to newly released version " $RELEASE_VERSION
+            rm -rf stable
+            ln -s $RELEASE_VERSION stable
+            git add stable
+            git commit -m "redirect stable to new version $RELEASE_VERSION"
+            cd ..
+
+      - name: Push changes
+        uses: ad-m/github-push-action@master
```

Comment:
same here with master.

Also what does this action actually do? I think you could just do a `git push` in the previous step right? Possibly we will have to set the read/write permission in the repo settings:

https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#preventing-github-actions-from-creating-or-approving-pull-requests

Reply:
have set v0.6.0
Shall we for now stay with the github action? It works fine...